lib/sqlalchemy/sql/schema.py

```diff
@@ -5664,3 +5707,239 @@ def _copy(self, **kw: Any) -> Identity:
         )
 
         return self._schema_item_copy(i)
+
+
+class Period(SchemaItem):
+    """Defines a period construction, i.e. ``PERIOD FOR period_name
+    (start_col, end_col)`` syntax.
+
+    .. versionadded:: TODO
+    """
+
+    __visit_name__ = "period"
+
+    table: Table
+    start: Column
+    end: Column
+    system: bool
+
+    def __init__(
+        self,
+        name: str,
+        start: _DDLColumnArgument,
+        end: _DDLColumnArgument,
+        system: Optional[bool] = False,
+        primary_key: Optional[bool] = False,
+    ) -> None:
+        """Construct a ``PERIOD FOR`` DDL construct on two columns.
+
+        :param name:
+            The name to give the period.
+
+        :param start:
+            A target column that defines the period's start. A
+            :class:`_schema.Column` object or a column name as a string.
+
+        :param end:
+            A target column that defines the period's start. A
+            :class:`_schema.Column` object or a column name as a string.
+
+        :param system:
+            Indicates that the ``Period`` is implicit, i.e. should not be
+            generated in DDL.
+
+        :param system:
+            Indicates that the ``Period`` is implicit, i.e. not generated in
+            DDL.
+
+        :param primary_key:
+            Whether this ``Period`` should be included in the table's primary
+            key.
+        """
+        if not system and (start is None) or (end is None):
+            exc.ArgumentError(
+                "Period object requires both start"
+                " and end columns must be specified"
+            )
+        self.name = name
+        self.start = coercions.expect(roles.DDLExpressionRole, start)
+        self.end = coercions.expect(roles.DDLExpressionRole, end)
+        self.system = system
+        self.primary_key = primary_key
+
+    @property
+    def key(self) -> str:
+        return self.name
+
+    def _get_and_validate_column(
+        self, table: Table, column: Union[str, Column]
+    ) -> Column:
+        """Accept a string or Column object, validate it is in metadata and
+        return the Table object."""
+        if isinstance(column, Column):
+            if column.table is not table:
+                raise exc.ArgumentError(
+                    "Given column object '%s' does not belong to Table '%s'"
+                    % (column.name, table.description)
+                )
+            return column
+
+        col = table.c.get(str(column))
+
+        if col is None:
+            raise exc.ArgumentError(
+                "Cannot find column '%s' in Table '%s'"
```

Comment:
**Federico Caselli** (CaselIT) wrote:

this seems unnecessary.
I think it we could always ignore it, maybe adding a mention in the history_table docs

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3803